### PR TITLE
Terraria 1.4 tModLoader

### PR DESCRIPTION
**Pull request by rakles**
I reworked the tModLoader template to work with the latest tModLoader release for the 1.4 version of Terraria.
They changed the server structure which made it incompatible with the other template.
This template is also incompatible with the older releases of the tModLoader before the 1.4 release, hence the separate template.

### CODE REVIEW

File: terraria-1.4-tmodloader/terraria-1.4-tmodloader.json [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+  },
+  "environment": {
+    "type": "tty"
+  },
+  "supportedEnvironments": [
+    {
+      "type": "tty"
+    }
+  ],
+  "requirements": {}
```
Comment (LordRalex) [inline terraria-1.4-tmodloader/terraria-1.4-tmodloader.json]: Please add any requirements needed for this (like what binaries are needed on the server, such as unzip)

File: terraria-1.4-tmodloader/terraria-1.4-tmodloader.json [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    "post": [],
+    "environmentVars": {}
+  },
+  "data": {
+    "abr": {
+      "type": "string",
+      "desc": "Sets the announcementbox text messaging range in pixels, -1 for serverwide announcements.",
+      "display": "announcementbox-range",
+      "required": true,
+      "value": "-1"
```
Comment (LordRalex) [inline terraria-1.4-tmodloader/terraria-1.4-tmodloader.json:64]: Make any number fields a "type": "integer"

**Review by LordRalex — CHANGES_REQUESTED** [review]

File: terraria-1.4-tmodloader/terraria-1.4-tmodloader.json (hunk 1 above)
Comment (rakles) [reply terraria-1.4-tmodloader/terraria-1.4-tmodloader.json]: Added all requirements in https://github.com/PufferPanel/templates/pull/164/commits/f7866d721e4e073d351b147614741dc4fcd72233

File: terraria-1.4-tmodloader/terraria-1.4-tmodloader.json (hunk 2 above)
Comment (rakles) [reply terraria-1.4-tmodloader/terraria-1.4-tmodloader.json:64]: Changed any number value to integer in https://github.com/PufferPanel/templates/pull/164/commits/04d2fac6a53d24b06a83d43db7242634708357ae

**Comment by LordRalex** [comment]
Please rebase this PR to v2.6 and follow our formatting rules:
https://github.com/PufferPanel/templates/blob/v2.6/README.md#formattng-rules

Please also confirm there is not a corrected template in the new branch.

**Pull request closed without merging**